File: src/main/java/com/thegame/business/model/ResourceType.java
package com.thegame.business.model;

import jakarta.persistence.Column;
import jakarta.persistence.Entity;
import jakarta.persistence.Id;
import jakarta.persistence.Table;

@Entity
@Table(name = "resource_types", schema = "data")
public class ResourceType {

    @Id
    @Column(name = "resource_type_id")
    private Long resourceTypeId;

    @Column(name = "resource_name")
    private String resourceName;

    public Long getResourceTypeId() {
        return resourceTypeId;
    }

    public void setResourceTypeId(Long resourceTypeId) {
        this.resourceTypeId = resourceTypeId;
    }

    public String getResourceName() {
        return resourceName;
    }

    public void setResourceName(String resourceName) {
        this.resourceName = resourceName;
    }
}

File: src/main/java/com/thegame/model/ResourceType.kt
package com.thegame.model

import jakarta.persistence.Column
import jakarta.persistence.Entity
import jakarta.persistence.Id
import jakarta.persistence.Table

@Entity
@Table(name = "resource_types", schema = "data")
class ResourceType {

    @Id
    @Column(name = "resource_type_id")
    var resourceTypeId: Long? = null

    @Column(name = "resource_name")
    var resourceName: String? = null

}